straxen/plugins/afterpulse_processing.py

```diff
@@ -0,0 +1,345 @@
+                       flip=True)
+ 
+        # find all hits
+        hits = strax.find_hits(r,
+                               min_amplitude = self.hit_thresholds,
+                               min_height_over_noise = self.config['hit_min_height_over_noise'],
+                              )
+        
+        # sort hits first by record_i, then by time
+        hits_sorted = np.sort(hits, order=('record_i', 'time'))
+        
+        # find LED hits and afterpulses within the same WF
+        hits_ap = find_ap(hits_sorted,
+                          r,
+                          LED_window_left=self.config['LED_window_left'],
+                          LED_window_right=self.config['LED_window_right'],
+                          hit_left_extension=self.config['hit_left_extension'],
+                          hit_right_extension=self.config['hit_right_extension'],
+                         )
+    
+        hits_ap['area_pe'] = hits_ap['area'] * self.to_pe[hits_ap['channel']]
+        hits_ap['height_pe'] = hits_ap['height'] * self.to_pe[hits_ap['channel']]
+        
+        return hits_ap
+
+
+@export
+def find_ap(hits, records, LED_window_left, LED_window_right, hit_left_extension, hit_right_extension):
+    buffer = np.zeros(len(hits), dtype=dtype_ap)
+    if not len(hits):
+        return buffer
+    res = _find_ap(hits, records, LED_window_left, LED_window_right, hit_left_extension, hit_right_extension, buffer=buffer)
+    return res
+            
+
+@numba.jit(nopython=True, nogil=True, cache=True)
+def _find_ap(hits, records, LED_window_left, LED_window_right, hit_left_extension, hit_right_extension, buffer=None):
+    # hits need to be sorted by record_i, then time!
+    
+    offset = 0
+    
+    is_LED = False
+    t_LED = None
+    
+    prev_record_i = hits[0]['record_i']
+    record_data = records[prev_record_i]['data']
+    record_len = records[prev_record_i]['length']
+    baseline_fpart = records[prev_record_i]['baseline'] % 1
+    
+    for h_i, h in enumerate(hits):
+        
+        if h['record_i'] > prev_record_i:
+            # start of a new record
+            is_LED = False
+            # only increment buffer if the old one is not empty! this happens when no (LED) hit is found in the previous record
+            if not buffer[offset]['time'] == 0:
+                offset += 1
+            prev_record_i = h['record_i']
+            record_data = records[prev_record_i]['data']
+            baseline_fpart = records[prev_record_i]['baseline'] % 1
+        
+        res = buffer[offset]
+
+            
+        if h['left'] < LED_window_left:
+            # if hit is before LED window: discard
+            continue
+            
+        if (h['left'] >= LED_window_left) & (h['left'] < LED_window_right):
+            # hit is in LED window
+            if not is_LED:
+                # this is the first hit in the LED window
+                res['time'] = h['time']
```

Comment:
You have to be care full here. Since you specify `length` to be `res['right_integration'] - res['left_integration']` you have to updated time by `h['time'] - hit_left_extension*h['dt'] `. Otherwise time + length * dt will not cover the correct interval.